From 9837008e995022dbe35b0452b6bc1fb72d32bb4b Mon Sep 17 00:00:00 2001
From: Feng Kan <fkan@apm.com>
Date: Thu, 23 Oct 2014 18:24:18 -0700
Subject: [PATCH 05/10] arm64: dts: Add X-Gene reboot driver dts node
Git-Commit: 5c3a87e363c09242541620a777d5b73e89b6c245
Patch-mainline: v4.4-rc1
References: bnc#902632

Add X-Gene platform reboot driver dts node.

[dhdang: changelog]
Signed-off-by: Feng Kan <fkan@apm.com>
Signed-off-by: Duc Dang <dhdang@apm.com>
Signed-off-by: Alexander Graf <agraf@suse.de>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>

---
Changed the order of inclusion.

 arch/arm64/boot/dts/apm/apm-storm.dtsi | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm64/boot/dts/apm/apm-storm.dtsi b/arch/arm64/boot/dts/apm/apm-storm.dtsi
index 295c72d..e0295e1 100644
--- a/arch/arm64/boot/dts/apm/apm-storm.dtsi
+++ b/arch/arm64/boot/dts/apm/apm-storm.dtsi
@@ -103,6 +103,11 @@
 		ranges;
 		dma-ranges = <0x0 0x0 0x0 0x0 0x400 0x0>;
 
+		scu: system-clk-controller@17000000 {
+			compatible = "apm,xgene-scu","syscon";
+			reg = <0x0 0x17000000 0x0 0x400>;
+		};
+
 		clocks {
 			#address-cells = <2>;
 			#size-cells = <2>;
@@ -469,6 +474,13 @@
 			clocks = <&pcie4clk 0>;
 		};
 
+		reboot: reboot@17000014 {
+			compatible = "syscon-reboot";
+			regmap = <&scu>;
+			offset = <0x14>;
+			mask = <0x1>;
+		};
+
 		serial0: serial@1c020000 {
 			status = "disabled";
 			device_type = "serial";
-- 
2.1.0

